tests/integration/models/reddit/test_rules.py

```diff
@@ -0,0 +1,139 @@
+from unittest import mock
+import pytest
+
+from praw.models import Rule
+
+from ... import IntegrationTest
+
+
+class TestRule(IntegrationTest):
+    @property
+    def subreddit(self):
+        return self.reddit.subreddit(pytest.placeholders.test_subreddit)
+
+    def test_add_rule(self):
+        self.reddit.read_only = False
+        with self.recorder.use_cassette("TestRule.test_add_rule"):
+            rule = self.subreddit.rules.add(
+                "PRAW Test",
+                "all",
+                description="Test by PRAW",
+                violation_reason="PTest",
+            )
+            assert rule.short_name == "PRAW Test"
+            assert rule.kind == "all"
+            assert rule.description == "Test by PRAW"
+            assert rule.violation_reason == "PTest"
+
+    def test_add_rule_without_violation_reason(self):
+        self.reddit.read_only = False
+        with self.recorder.use_cassette(
+            "TestRule.test_add_rule_without_violation_reason"
+        ):
+            rule = self.subreddit.rules.add("PRAW Test 2", "comment")
+            assert rule.short_name == "PRAW Test 2"
+            assert rule.kind == "comment"
+            assert rule.description == ""
+            assert rule.violation_reason == "PRAW Test 2"
+
+    @mock.patch("time.sleep", return_value=None)
+    def test_delete_rule(self, _):
+        self.reddit.read_only = False
+        with self.recorder.use_cassette("TestRule.test_delete_rule"):
+            rule = self.subreddit.rules[-1]
+            rule.delete()
+
+    def test_get_rules(self):
+        with self.recorder.use_cassette("TestRule.test_get_rules"):
+            for rule in self.subreddit.rules:
+                assert isinstance(rule, Rule)
+
+    def test_get_rule_string(self):
+        with self.recorder.use_cassette("TestRule.test_get_rule_string"):
+            rule = self.subreddit.rules["PRAW Test"]
+            assert isinstance(rule, Rule)
+            assert rule.short_name == "PRAW Test"
+
+    def test_get_rule_int(self):
+        with self.recorder.use_cassette("TestRule.test_get_rule_int"):
+            assert isinstance(self.subreddit.rules[0], Rule)
+
+    def test_get_rule_negative_int(self):
+        with self.recorder.use_cassette("TestRule.test_get_rule_int"):
+            assert isinstance(self.subreddit.rules[-1], Rule)
+
+    @mock.patch("time.sleep", return_value=None)
+    def test_reorder_rules(self, _):
+        self.reddit.read_only = False
+        with self.recorder.use_cassette("TestRule.test_reorder_rules"):
+            rule_list = list(self.subreddit.rules)
+            reordered = rule_list[2:3] + rule_list[0:2] + rule_list[3:]
+            rule_info = {rule.short_name: rule for rule in rule_list}
+            self.subreddit.rules.reorder(reordered)
+            for rule in self.subreddit.rules:
```

Comment:
As with the delete above, can we re-fetch and then verify that the new order has been set?

Reply:
That would also be a good idea.